### ISSUE TITLE
Abnormal disconnections lock out players from rejoining

### ISSUE THREAD

**Issue by sg4e** [riichinomics/autotable#28]
When a player experiences an abnormal disconnection (such as from an Internet outage or a lag spike) while making a move, the game can enter a state where the player's seat is still marked as occupied, and when reconnecting, the player is unable to reclaim their seat. In this state, the game cannot be completed, and a new room must be created with the current hand lost.

The upstream repo has fixed this scenario by implementing an "Unseat" button, see commits:
- https://github.com/pwmarcz/autotable/commit/87fc565b3629dbd97778594a15d4399ff9c75dbc
- https://github.com/pwmarcz/autotable/commit/0d96e47427ed86906029f4eb112a53256a91596e

Is it possible to merge these commits to solve this issue?

**Comment by riichinomics** [comment]
Yes, this is a well known issue to me. We have this happen regularly. Eventually the server removes the inactive player. Unfortunately this fork has diverged from the upstream significantly and I'm not sure if we can simply cherry pick those commits, but I will look into resolving this issue.

**Comment by 5zlotych** [comment]
Hi, I noticed that problem yesterday and it was pretty frustrating. Just a little bump up so you remember about that, It'd be great if it got fixed 😔

**Comment by riichinomics** [comment]
I've added some heartbeat logic to the server to clear dead connections. Hopefully it should resolve this issue, but I will keep it open for now.
